scripts/get-cert-manager.sh

```diff
@@ -1,13 +1,7 @@
 #!/bin/bash
 
 # cert-manager is ready for CRD objects when this condition is "True"
-CERT_READY=$(kubectl get cert/cert-manager-webhook-webhook-tls -n  cert-manager -o jsonpath="{.status.conditions[0].status}")
-WEBHOOK_READY=$(kubectl get deploy/cert-manager-webhook -n cert-manager -o jsonpath="{.status.conditions[0].status}")
+kubectl rollout status cert/cert-manager-webhook-webhook-tls -n  cert-manager --timeout=10m
+kubectl rollout status deploy/cert-manager-webhook -n cert-manager --timeout=10m
 
-if [ "$CERT_READY" = "True" ]
-then
-    if [ "$WEBHOOK_READY" = "True" ]
-    then
-        echo -n True
-    fi
-fi
+sleep 2
```

Comment:
Why is this needed?

Reply:
While testing, I found it complaining `resource cert` not found in the next steps, even tough it shows deployment is ready. 

Same thing happened with `gateway` as well. So I have added a two second sleep there as well. In gateway rollout check it shows rolled out but when tries to deploy the function it shows an error.

Reply:
What error did the gateway give?

With the previous code if it found an error applying the resources it would retry.

Adding an arbitrary sleep is bound to break at some point when the network is slower or some other condition exists. I'd rather it worked all the time.

Do you have any other suggestions?

Reply:
This is the error, I was getting in case of cert manager. 

<details>
Attempt -1 :
<summary>Output</summary>

```
exec:  kubectl apply -f tmp/generated-tls-issuer-prod.yml
res:
2019/04/08 07:10:55 0  Error from server (InternalError): error when creating "tmp/generated-tls-issuer-prod.yml": Internal error occurred: failed calling webhook "clusterissuers.admission.certmanager.k8s.io": the server is currently unable to handle the request

exec:  kubectl apply -f tmp/generated-tls-issuer-staging.yml
res:
2019/04/08 07:10:55 0  Error from server (InternalError): error when creating "tmp/generated-tls-issuer-staging.yml": Internal error occurred: failed calling webhook "clusterissuers.admission.certmanager.k8s.io": the server is currently unable to handle the request

exec:  kubectl apply -f tmp/generated-tls-wildcard-domain-cert.yml
res:
2019/04/08 07:10:55 0  Error from server (InternalError): error when creating "tmp/generated-tls-wildcard-domain-cert.yml": Internal error occurred: failed calling webhook "certificates.admission.certmanager.k8s.io": the server is currently unable to handle the request

exec:  kubectl apply -f tmp/generated-tls-auth-domain-cert.yml
res: certificate.certmanager.k8s.io/auth-system-vs.team-serverless.xyz created

2019/04/08 07:10:56 0 certificate.certmanager.k8s.io/auth-system-vs.team-serverless.xyz created
```
</details>

<details>
Attempt - 2 
```
<summary>Output</summary>

2019/04/08 07:19:05 0 ingress.extensions/openfaas-auth-ingress created

exec:  kubectl apply -f tmp/generated-tls-issuer-prod.yml
res:
2019/04/08 07:19:05 0  Error from server (InternalError): error when creating "tmp/generated-tls-issuer-prod.yml": Internal error occurred: failed calling webhook "clusterissuers.admission.certmanager.k8s.io": the server is currently unable to handle the request

exec:  kubectl apply -f tmp/generated-tls-issuer-staging.yml
res:
2019/04/08 07:19:05 0  Error from server (InternalError): error when creating "tmp/generated-tls-issuer-staging.yml": Internal error occurred: failed calling webhook "clusterissuers.admission.certmanager.k8s.io": the server is currently unable to handle the request

exec:  kubectl apply -f tmp/generated-tls-wildcard-domain-cert.yml
res:
2019/04/08 07:19:06 0  Error from server (InternalError): error when creating "tmp/generated-tls-wildcard-domain-cert.yml": Internal error occurred: failed calling webhook "certificates.admission.certmanager.k8s.io": the server is currently unable to handle the request

exec:  kubectl apply -f tmp/generated-tls-auth-domain-cert.yml
res:
2019/04/08 07:19:06 0  Error from server (InternalError): error when creating "tmp/generated-tls-auth-domain-cert.yml": Internal error occurred: failed calling webhook "certificates.admission.certmanager.k8s.io": the server is currently unable to handle the request
```
</details>

cc @alexellis 

I completely agree if network is slow the this sleep will also break.

Reply:
A timeout is not suitable for the reasons given above.

The recommendation I had from JetStack was to create a temporary certificate in a retry loop until it passes / is accepted. 

See this section - https://docs.cert-manager.io/en/latest/getting-started/install.html#verifying-the-installation

Reply:
Should we remove the timeout and create a different issue to include the retry with temporary certificate  logic ?

Reply:
I don't think that an issue is needed. You can go ahead and do that through this PR